### DISCUSSION TITLE
Fenced code blocks not generating correct HTML

### DISCUSSION THREAD

**Discussion by ryangjchandler** [spatie/laravel-markdown#1] [Q&A] [answered]
I'm using this package to render Markdown with Shiki highlighting. Unfortunately, any fenced code blocks aren't being highlighted. They're not even being wrapped in `pre` or `code` tags.

For reference, here is my `markdown` function:

```php
function markdown(string $markdown, string $cacheKey = null) {
    /** @var \Spatie\LaravelMarkdown\MarkdownRenderer $converter */
    $converter = app(MarkdownRenderer::class)
        ->highlightTheme('github-light');

    if ($cacheKey && app()->environment('production')) {
        $converter->cacheStoreName($cacheKey);
    } else {
        $converter->cacheStoreName(false);
    }

    return $converter->toHtml($markdown);
}
```

**Edit:**

If I disable code highlighting using `$converter->highlightCode(false)`, my `<pre>` and `<code>` tags return but there's no code highlighting of course.

**Comment by ryangjchandler** [comment, marked as answer]
Okay, interesting development. It appears that it works via `artisan serve` but not when serving through Valet. I believe the `getenv('PATH')` isn't returning anything, therefore `ExecutableFinder` can't find the `node` binary.

Will have to dig further but it looks like an issue with my Valet setup. Will close this.

**Reply by edgrosvenor** [reply]
It might not just be your Valet setup. I'm having the same problem and even using `artisan serve` isn't solving it.

**Reply by edgrosvenor** [reply]
Nope. I was wrong. It's your Valet setup. I got it working in Valet and with `artisan serve`.

**Reply by myurtsev** [reply]
Same problem, without Valet. It's working locally, but not on production.

**Reply by ClintWinter** [reply]
I'm having the same issue. Not working on production for me either.

**Reply by hilmanski** [reply]
hi any update on your solution? I'm using sail for development it worked, but not on production

**Comment by dilab** [comment]
@ryangjchandler what did you do to fix the issue?

**Reply by myurtsev** [reply]
@dilab did you find a solution for this in the mean time? Anyone?

**Comment by myurtsev** [comment]
It seems like this (also) occurs when you use NVM. The package can't find the node executable, so you need to create a symlink for it.

**Reply by hilmanski** [reply]
do you found a solution for production on your case @myurtsev  ? thanks in advance
